tests/scalers/gitlab_runner/gitlab_runner_test.go

```diff
@@ -0,0 +1,503 @@
+	MinReplicaCount           string
+	MaxReplicaCount           string
+	Pat                       string
+	GitlabAPIURL              string
+	ProjectID                 string
+	TargetPipelineQueueLength string
+}
+
+const (
+	secretTemplate = `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  personalAccessToken: {{.Pat}}
+`
+	triggerAuthTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: gitlab-trigger-auth
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+    - parameter: personalAccessToken
+      name: {{.SecretName}}
+      key: personalAccessToken
+`
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: gitlab-runner
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: gitlab-runner
+  template:
+    metadata:
+      labels:
+        app: gitlab-runner
+    spec:
+      terminationGracePeriodSeconds: 90
+      containers:
+      - name: gitlab-runner
+        image: busybox
+        command: ["/bin/sh", "-c", "trap 'echo SIGTERM received; exit 0' SIGTERM; while true; do sleep 30; done"]
+`
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 15
+  cooldownPeriod: 5
+  triggers:
+  - type: gitlab-runner
+    metadata:
+      gitlabAPIURL: {{.GitlabAPIURL}}
+      projectID: "{{.ProjectID}}"
+      targetPipelineQueueLength: "{{.TargetPipelineQueueLength}}"
+      labels: "e2eSOtester"
+    authenticationRef:
+     name: gitlab-trigger-auth
+`
+)
+
+func TestScaler(t *testing.T) {
+	// setup
+	t.Log("--- setting up ---")
+
+	t.Log("--- deleting all projects ---")
+	err := deleteAllUserProjects(gitlabBaseURL)
+	require.NoError(t, err)
+
+	t.Log("--- creating new project ---")
+	projectID, err := createNewProject(gitlabBaseURL)
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("--- cleanup project ---")
+		err := deleteRepo(gitlabBaseURL, projectID)
+		require.NoError(t, err)
+	}()
+
+	t.Log("--- add ci file ---")
+	err = commitFile(gitlabBaseURL, ciFileContent, projectID, ".gitlab-ci.yml", true)
+	require.NoError(t, err)
+
+	// Create kubernetes resources
+	t.Log("--- create kubernetes resources ---")
+	kc := GetKubernetesClient(t)
+	data, templates := getTemplateData(projectID)
+	CreateKubernetesResources(t, kc, testNamespace, data, templates)
+
+	WaitForPodCountInNamespace(t, kc, testNamespace, minReplicaCount, 60, 2)
+
+	// test scaling Scaled Object
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+	testSONotActivated(t, kc)
+
+	testSOScaleOut(t, kc, projectID)
+	testSOScaleIn(t, kc, projectID)
```

Comment:
Let's use the same name as the majority of the tests:
```suggestion
	testActivation(t, kc)

	testScaleOut(t, kc, projectID)
	testScaleIn(t, kc, projectID)
```